core/src/test/java/com/datastax/oss/driver/internal/core/metadata/schema/queries/DefaultSchemaQueriesFactoryTest.java

```diff
@@ -0,0 +1,130 @@
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import java.util.concurrent.CompletableFuture;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class DefaultSchemaQueriesFactoryTest {
+
+  enum Expected {
+    CASS_21(Cassandra21SchemaQueries.class),
+    CASS_22(Cassandra22SchemaQueries.class),
+    CASS_3(Cassandra3SchemaQueries.class),
+    CASS_4(Cassandra4SchemaQueries.class);
+
+    final Class<? extends SchemaQueries> clz;
+
+    Expected(Class<? extends SchemaQueries> clz) {
+      this.clz = clz;
+    }
+
+    public Class<? extends SchemaQueries> getClz() {
+      return clz;
+    }
+  }
+
+  private static When<Expected> whenCassandra =
+      When.<Expected>builder()
+          .args("2.1.0")
+          .expect(Expected.CASS_21)
+          .args("2.2.0")
+          .expect(Expected.CASS_22)
+          .args("2.2.1")
+          .expect(Expected.CASS_22)
+          /* Not a real version, just documenting behaviour of existing impl */
+          .args("2.3.0")
+          .expect(Expected.CASS_22)
+          /* We now return you to real versions */
+          .args("3.0.0")
+          .expect(Expected.CASS_3)
+          .args("3.0.1")
+          .expect(Expected.CASS_3)
+          .args("3.1.0")
+          .expect(Expected.CASS_3)
+          .args("4.0.0")
+          .expect(Expected.CASS_4)
+          .args("4.0.1")
+          .expect(Expected.CASS_4)
+          .args("4.1.0")
+          .expect(Expected.CASS_4)
+          .build();
+
+  private static When<Expected> whenDse =
+      When.<Expected>builder()
+          /* DSE 6.0.0 */
+          .args("4.0.0.2284")
+          .expect(Expected.CASS_3)
+          /* DSE 6.0.1 */
+          .args("4.0.0.2349")
+          .expect(Expected.CASS_3)
+          /* DSE 6.0.2 moved to DSE version (minus dots) in an extra element */
+          .args("4.0.0.602")
+          .expect(Expected.CASS_3)
+          /* DSE 6.7.0 continued with the same idea */
+          .args("4.0.0.670")
+          .expect(Expected.CASS_3)
+          /* DSE 6.8.0 does the same */
+          .args("4.0.0.680")
+          .expect(Expected.CASS_3)
+          .build();
+
+  @DataProvider(format = "%m %p[1] => %p[0]")
+  public static Iterable<Iterable<Object>> expected() {
+
+    return whenCassandra.merge(whenDse);
+  }
+
+  @Test
+  @UseDataProvider("expected")
+  public void should_return_correct_schema_queries_impl(Expected expected, String version) {
+
+    final Node mockNode = mock(Node.class);
+    when(mockNode.getCassandraVersion()).thenReturn(Version.parse(version));
+
+    DefaultSchemaQueriesFactory factory = buildFactory();
+    SchemaQueries queries =
+        factory.newInstance(mockNode, mock(DriverChannel.class), mock(CompletableFuture.class));
```

Comment:
You may want to add `SupressWarninig("unchecked")` for this one:
`mock(CompletableFuture.class)`

Reply:
Agreed.  I'd noticed this in Eclipse and hadn't actually put in the fix yet.  Will add in the next round of updates.